AdyenActions/UI/View Controllers/DelegatedAuthentication/DARegistrationViewController.swift

```diff
@@ -0,0 +1,133 @@
+        self.notNowHandler = notNowHandler
+        super.init(nibName: nil, bundle: Bundle(for: DARegistrationViewController.self))
+        registrationView.delegate = self
+        if #available(iOS 13.0, *) {
+            isModalInPresentation = true
+        }
+    }
+    
+    @available(*, unavailable)
+    internal required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    override internal func viewDidLoad() {
+        super.viewDidLoad()
+        buildUI()
+        configureDelegateAuthenticationView()
+        view.backgroundColor = style.backgroundColor
+        configureProgress()
+        configureDelegateAuthenticationView()
+    }
+    
+    private func configureDelegateAuthenticationView() {
+        registrationView.titleLabel.text = localizedString(.threeds2DARegistrationTitle, localizationParameters)
+        registrationView.descriptionLabel.text = localizedString(.threeds2DARegistrationDescription, localizationParameters)
+        registrationView.firstButton.title = localizedString(.threeds2DARegistrationPositiveButton, localizationParameters)
+        registrationView.secondButton.title = localizedString(.threeds2DARegistrationNegativeButton, localizationParameters)
+        configureProgress()
+    }
+
+    private func configureProgress() {
+        let timeout: TimeInterval = 90.0
+        registrationView.progressText.text = timeLeft(timeInterval: timeout)
+        timeoutTimer = ExpirationTimer(
+            expirationTimeout: timeout,
+            onTick: { [weak self] in
+                self?.registrationView.progressView.progress = Float($0 / timeout)
+                self?.registrationView.progressText.text = self?.timeLeft(timeInterval: $0)
+            },
+            onExpiration: { [weak self] in
+                self?.firstButtonTapped()
+            }
+        )
+        timeoutTimer?.startTimer()
+    }
+
+    private func timeLeft(timeInterval: TimeInterval) -> String {
+        String(format: localizedString(.threeds2DARegistrationTimeLeft, localizationParameters), timeInterval.adyen.timeLeftString() ?? "0")
+    }
+    
+    private func buildUI() {
+        containerView.addSubview(registrationView)
+        view.addSubview(containerView)
+
+        containerView.translatesAutoresizingMaskIntoConstraints = false
+        registrationView.translatesAutoresizingMaskIntoConstraints = false
+        
+        containerView.adyen.anchor(inside: view.safeAreaLayoutGuide)
+
+        let constraints = [
+            registrationView.topAnchor.constraint(equalTo: containerView.topAnchor),
+            registrationView.bottomAnchor.constraint(equalTo: containerView.bottomAnchor),
+            
+            registrationView.leadingAnchor.constraint(equalTo: containerView.leadingAnchor),
+            registrationView.trailingAnchor.constraint(equalTo: containerView.trailingAnchor)
+        ]
+        NSLayoutConstraint.activate(constraints)
```

Comment:
We have [helper](https://github.com/Adyen/adyen-ios/blob/develop/Adyen/Helpers/UIViewConstraintsHelper.swift#L18-L21) for this:

```suggestion
     registrationView.adyen.anchor(inside: containerView)
```

Reply:
Convenient. Thanks. Changed it.